apps/web/src/lib/aggregate-energy.ts

```diff
@@ -0,0 +1,109 @@
+export function getAggregatedEnergy(data: Record<string, string | number | undefined>[], aggregationType: string | undefined) {
+
+    if(aggregationType == null) {
+        aggregationType = "hour"
+    }
+
+    switch (aggregationType) {
+        case 'hour':
+            return handleHourOption(data);
+        case 'day':
+            return handleDayOption(data);
+        case 'month':
+            return handleMonthOption(data);
+        case 'year':
+            return handleYearOption(data);
+    }
+    return data
+};
+
+const handleHourOption = (data) => {
+    const energyConsumptionPerHour = [];
+
+    for (const value in data) {
+        const timestampParts = data[value].timestamp.split(' ');
+        const date = timestampParts[1] + ' ' + timestampParts[2] + ' ' + timestampParts[3];
+        const hour = timestampParts[4].split(':')[0];
+
+        const hourKey = date + ' ' + hour;
+
+        if (energyConsumptionPerHour[hourKey]) {
+            energyConsumptionPerHour[hourKey].energy += data[value].energy;
+        } else {
+            energyConsumptionPerHour[hourKey] = {
+                energy: data[value].energy,
+                timestamp: data[value].timestamp
+            };
+        }
+    }
+
+    return Object.values(energyConsumptionPerHour);
+}
+
+
+const handleDayOption = (data) => {
+    const energyConsumptionPerDay = [];
+
+    for (const value in data) {
+        const date = 
+            data[value].timestamp.split(' ')[1] + ' ' + 
```

Comment:
Wie bei `handleHourOption`
```suggestion
            const timestampParts = data[value].timestamp.split(' ');
            timestampParts[1] + ' ' + 
```